gallery/src/components/NxToast/NxToastFullPageVariationsExample.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { useState, ReactElement } from 'react';
+import {reject, propEq} from 'ramda';
+import {
+  NxGlobalSidebarNavigation,
+  NxGlobalSidebarNavigationLink,
+  NxPageSidebar,
+  NxButton,
+  NxFontAwesomeIcon,
+  NxBackButton,
+  NxStatefulGlobalSidebar,
+  NxPageMain,
+  NxPageTitle,
+  NxH1,
+  NxToastContainer,
+  NxToast,
+  NxSuccessAlert,
+  NxErrorAlert,
+  NxInfoAlert,
+  NxWarningAlert,
+  NxTile,
+  NxP
+} from '@sonatype/react-shared-components';
+import {
+  faArrowLeft,
+  faArrowRight,
+  faLink,
+  faQuestionCircle,
+  faBell,
+  faCog,
+  faUserCircle
+} from '@fortawesome/free-solid-svg-icons';
+
+const sidebarLogoPath = require('../../assets/images/logo-plaid-villain-text.png');
+
+const alerts = {
+  success: NxSuccessAlert,
+  error: NxErrorAlert,
+  info: NxInfoAlert,
+  warning: NxWarningAlert
+};
```

Comment:
Since we are leaving it to the consuming devs to specify the chosen alerts to be wrapped in NxToast, I'm wondering if defining the alerts within an alert object for the dev to choose from is ok.

Reply:
> I think the heart of the question here is whether we imagine the calling code storing actual toast React nodes in the state, or just storing some indication of the toast type in the state and then constructing the toast JSX from that when converting from state to UI. The latter is probably cleaner. In that scenario, there'd still be a mapping object like this but I think it would be cleaner if it was referenced when rendering the toasts, not in `addToast`.

I implemented a working copy here: [c1e07de](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/c1e07de714219ad666458562a5b9e494da737abd) . I wanted to try and use <NxAlert> and style it with nx-alert--${type} but Alert takes an icon, which would still require a nested ternery to assign the appropriate icon (and lose some aria-roles). I'm wondering if this implementation of mapping is clean enough?

Reply:
> > I think the heart of the question here is whether we imagine the calling code storing actual toast React nodes in the state, or just storing some indication of the toast type in the state and then constructing the toast JSX from that when converting from state to UI. The latter is probably cleaner. In that scenario, there'd still be a mapping object like this but I think it would be cleaner if it was referenced when rendering the toasts, not in `addToast`.
> 
> I implemented a working copy here: [c1e07de](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/c1e07de714219ad666458562a5b9e494da737abd) . I wanted to try and use `NxAlert` and style it with `nx-alert--${type}` but Alert takes an icon, which would still require a nested ternery to assign the appropriate icon (and lose some aria-roles). I'm wondering if this implementation of mapping is clean enough?

Scratch this comment for now. Misunderstood a tad and refactoring now.